fix(store,world): minor config validation fixes (#2517)
latticexyz/mud: .changeset/dry-toes-lay.md

---
"@latticexyz/store": patch
"@latticexyz/world": patch
---

> Minor fixes to validations of the new config:

Author: ```suggestion
Minor fixes to validations of the config:
```

> Minor fixes to validations of the new config:

Reviewer: ```suggestion
Minor fixes to TS validation of the config:
```

> Minor fixes to validations of the new config:

Author: ```suggestion
Minor fixes to config input validations:
```
